docs: Update the README with the branding procedure from the docs PR (#155)
che-incubator/che-code: README.md

$ git remote add upstream-code https://github.com/microsoft/vscode
```

> #### Pulling changes from the remote Code

Reviewer: "the remote Code" ? why not from https://github.com/microsoft/vscode ? 

```suggestion
#### Pulling changes from upstream https://github.com/microsoft/vscode
```

git fetch upstream-code main
```
3. `./rebase.sh`
4. Fix the conflicts or other errors. **Note**, that`./rebase.sh` script also applies the patches from the [`.rebase`](https://github.com/che-incubator/che-code/tree/main/.rebase) directory. Sometimes, it also requires some updates there.
5. Open a PR with your changes.

## Branding the UI

   <details>
          <summary>You can brand some of the UI elements in the Visual Studio Code - Open Source IDE with your corporate or product brand.</summary>

\

> This means first adding brand-related files to the forked IDE repository, then building a container image of the branded IDE, and finally adding a `che-editor.yaml` file to the project repository.

Reviewer: ```suggestion
There are three steps involved in rebranding the IDE. 
1\. Add brand-related files to the forked IDE repository, 
2\. Build a container image of the branded IDE, and 
3\. Add a `che-editor.yaml` file to the project repository.
```

	background-image: url(./che/icon.svg);
	width: 13px;
	height: 13px;
}
```

6\. Run the `/branding/branding.sh` script. The [branding.sh](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh) script searches for the branding resources in the [branding folder](https://github.com/che-incubator/che-code/tree/main/branding) and applies the changes.

```
$ ./branding/branding.sh
```
<!-- Currently, the [branding.sh](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh) script is not run automatically when building this project. It needs to be integrated into the build process of the [downstream branded project or product](https://github.com/redhat-developer/devspaces-images/blob/devspaces-3-rhel-8/devspaces-code/build/scripts/sync.sh#L96). -->

7\. Build the container image from the `/che-code/` directory and push the image to a container registry:

```
$ docker build -f build/dockerfiles/linux-musl.Dockerfile -t linux-musl-amd64 .

$ docker build -f build/dockerfiles/linux-libc.Dockerfile -t linux-libc-amd64 .

$ export DOCKER_BUILDKIT=1

$ docker build -f build/dockerfiles/assembly.Dockerfile -t vs-code-open-source:next .

> $ docker push <username>/vs-code-open-source:next

Reviewer: see https://github.com/che-incubator/che-code/pull/155/files#r1042554234

        endpoints:
          - name: che-code
            attributes:
              type: main
              cookiesAuthEnabled: true
              discoverable: false
              urlRewriteSupported: true
            targetPort: 3100
            exposure: public
            secure: false
            protocol: https
          - name: code-redirect-1
            attributes:
              discoverable: false
              urlRewriteSupported: true
            targetPort: 13131
            exposure: public
            protocol: http
          - name: code-redirect-2
            attributes:
              discoverable: false
              urlRewriteSupported: true
            targetPort: 13132
            exposure: public
            protocol: http
          - name: code-redirect-3
            attributes:
              discoverable: false
              urlRewriteSupported: true
            targetPort: 13133
            exposure: public
            protocol: http
      attributes:
        app.kubernetes.io/component: che-code-runtime
        app.kubernetes.io/part-of: che-code.eclipse.org
    - name: checode
      volume: {}
    - name: che-code-injector
      container:

>         image: quay.io/username/vs-code-open-source:next

Reviewer: see https://github.com/che-incubator/che-code/pull/155/files#r1042554234

        command: ["/entrypoint-init-container.sh"]
        volumeMounts:
          - name: checode
            path: /checode
        memoryLimit: 128Mi
        memoryRequest: 32Mi
        cpuLimit: 500m
        cpuRequest: 30m
```

> :grey_exclamation: In this example, `quay.io/username/vs-code-open-source:next` specifies the container image of a branded Visual Studio Code - Open Source that will be pulled at workspace creation.

Reviewer: since the likely outcome is an organization path, not a username path, should we suggest a more likely example?

```suggestion
:grey_exclamation: In this example, `quay.io/branding-organization/vs-code-open-source:next` specifies the container image of a branded Visual Studio Code - Open Source IDE that will be pulled at workspace creation.
```

Reviewer 2: Good catch.

4. Fix the conflicts or other errors. **Note**, that`./rebase.sh` script also apllies the patches from the [`.rebase`](https://github.com/che-incubator/che-code/tree/main/.rebase) directory. Sometimes, it also requires some updates there.
5. Open a PR with your changes.

## Branding the UI

   <details>
          <summary>You can brand some of the UI elements in the Visual Studio Code - Open Source IDE with your corporate or product brand.</summary>

\
This means first adding brand-related files to the forked IDE repository, then building a container image of the branded IDE, and finally adding a `che-editor.yaml` file to the project repository.

Here are some examples of the UI elements in Visual Studio Code - Open Source that you can brand:

* Browser tab title and icon
* The icon for the empty editor area when no editor is open
* The Status Bar commands
* The Status Bar icon
* The Get Started page
* The tab icon for the Get Started page

> * The application name in the **About** dialog

Reviewer 3: I think it makes sense:
- remove bold style for the `About`
or 
- use bold text for other items above (Status Bar, Get Started), like you do it below

<img width="808" alt="Screenshot 2022-11-26 at 10 16 02" src="https://user-images.githubusercontent.com/5676062/204079379-0d87ca3d-a6d3-446b-8d41-cbdce0ef8ad9.png">

wdyt?

Author: Thank you for spotting these, Roman! I've made those bold too.

*Example. `/branding/css/codicon.css`*

```css
span.codicon.codicon-eclipse-che  {
	background-image: url(./che/icon.svg);
	width: 13px;
	height: 13px;
}
```

6\. Run the `/branding/branding.sh` script. The [branding.sh](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh) script searches for the branding resources in the [branding folder](https://github.com/che-incubator/che-code/tree/main/branding) and applies the changes.

```
$ ./branding/branding.sh
```
<!-- Currently, the [branding.sh](https://github.com/che-incubator/che-code/blob/main/branding/branding.sh) script is not run automatically when building this project. It needs to be integrated into the build process of the [downstream branded project or product](https://github.com/redhat-developer/devspaces-images/blob/devspaces-3-rhel-8/devspaces-code/build/scripts/sync.sh#L96). -->

7\. Build the container image from the `/che-code/` directory and push the image to a container registry:

```
$ docker build -f build/dockerfiles/linux-musl.Dockerfile -t linux-musl-amd64 .

$ docker build -f build/dockerfiles/linux-libc.Dockerfile -t linux-libc-amd64 .

$ export DOCKER_BUILDKIT=1

> $ docker build -f build/dockerfiles/assembly.Dockerfile -t vs-code-open-source .

Reviewer 3: I just noticed an inaccuracy in our examples:

the `next` tag for the docker image is used in examples below, but by default docker sets `latest` tag at building images

so, please add `next` tag to the command
it should be:
`docker build -f build/dockerfiles/assembly.Dockerfile -t vs-code-open-source:next .`

Author: Fixed. Thank you.

        endpoints:
          - name: che-code
            attributes:
              type: main
              cookiesAuthEnabled: true
              discoverable: false
              urlRewriteSupported: true
            targetPort: 3100
            exposure: public
            secure: false
            protocol: https
          - name: code-redirect-1
            attributes:
              discoverable: false
              urlRewriteSupported: true
            targetPort: 13131
            exposure: public
            protocol: http
          - name: code-redirect-2
            attributes:
              discoverable: false
              urlRewriteSupported: true
            targetPort: 13132
            exposure: public
            protocol: http
          - name: code-redirect-3
            attributes:
              discoverable: false
              urlRewriteSupported: true
            targetPort: 13133
            exposure: public
            protocol: http
      attributes:
        app.kubernetes.io/component: che-code-runtime
        app.kubernetes.io/part-of: che-code.eclipse.org
    - name: checode
      volume: {}
    - name: che-code-injector
      container:

>         image: quay.io/username/vs-code-open-source:next <1>

Reviewer 3: I guess `<1>` should be removed:

<img width="538" alt="Screenshot 2022-11-26 at 09 45 33" src="https://user-images.githubusercontent.com/5676062/204078300-a37281c1-c165-4b11-b7b9-bfa0f978d302.png">

Author: Oh, thank you for spotting that! Fixing it.

> 4. Fix the conflicts or other errors. **Note**, that`./rebase.sh` script also apllies the patches from the [`.rebase`](https://github.com/che-incubator/che-code/tree/main/.rebase) directory. Sometimes, it also requires some updates there.

Reviewer: ```suggestion
4. Fix the conflicts or other errors. **Note**, that`./rebase.sh` script also applies the patches from the [`.rebase`](https://github.com/che-incubator/che-code/tree/main/.rebase) directory. Sometimes, it also requires some updates there.
```

`letterpress` is the icon for the empty editor area when no editor is open. You can provide different icon files for `light` and `dark` themes.

`remoteIndicatorCommands` is the names of commands provided by the [Eclipse Che Remote](https://github.com/che-incubator/che-code/blob/main/code/extensions/che-remote/package.nls.json) extension. Users can run these commands by clicking the **Status Bar**.

`workbenchConfigFilePath` is the relative path to `workbench-config.json`, which is explained in one of the next steps.

`codiconCssFilePath` is the relative path to `css/codicon.css`, which is explained in one of the next steps.

:grey_exclamation: The values defined in the `/branding/product.json` file override the [default values](https://github.com/che-incubator/che-code/blob/main/code/product.json).

3\. Add the icon files, which you specified in the `product.json` file in the previous step, to the repository.

4\. Create a `/branding/workbench-config.json` file with custom values.

*Example. `/branding/workbench-config.json`*

```
{
	"windowIndicator": {

> 		"label": "$(eclipse-che) Eclipse Che",
> 		"tooltip": "Eclipse Che"

Author: It's not clear to me what these two lines are exactly for: the Che instance or the VS Code?

```suggestion
		"label": "$(eclipse-che) Branded IDE",
		"tooltip": "Branded IDE"
```

Reviewer 3: it's responsible for this item:

<img width="238" alt="Screenshot 2022-11-25 at 21 46 07" src="https://user-images.githubusercontent.com/5676062/204049363-98ac1be4-0661-45c7-9f42-4026b1f9c86c.png">

the first part `$(eclipse-che)` - is rendered as the corresponding icon
the next one `Eclipse Che` - as the label
the last one - is tooltip

Reviewer 3: [here](https://github.com/eclipse-che/che-docs/pull/2467/files#diff-f0a52a1e5d88daf1e06a0693ef3331285f726a430a2f802c17b8f1b92324483c) you can see an example for the `Dev Spaces` status bar item when branding was applied

Author: Thank you, Roman.
Regarding `"label": "$(eclipse-che)`:
`(eclipse-che)`--is it `statusBarItem` from line 169? Is this a variable or how does it become a file name for the icon (or which icon file is it in `product.json`)?

Reviewer 3: > is it statusBarItem from line 169?

it's true

>  Is this a variable or how does it become a file name for the icon (or which icon file is it in product.json)?

<img width="1002" alt="Screenshot 2022-11-29 at 07 20 08" src="https://user-images.githubusercontent.com/5676062/204445432-8eebc78c-984f-4bb3-8c68-bd0c7054266d.png">

we defined `eclipse-che` codicon in the `workbench-config.json` file.
css file contains style with ending `eclipse-che`.
the file name is defined in the corresponding style.

<img width="425" alt="Screenshot 2022-11-29 at 07 24 47" src="https://user-images.githubusercontent.com/5676062/204446129-c09eb440-103d-429d-9879-5e831ddffdc4.png">

For example, for the `devspaces-code`:
- there is `$(devspaces)` codicon in [the workbench-config.json file](https://github.com/redhat-developer/devspaces-images/blob/ec9c9776a0f82093616ff0d96385273c8c1cbfbf/devspaces-code/branding/workbench-config.json#L3).
- [css file](https://github.com/redhat-developer/devspaces-images/blob/ec9c9776a0f82093616ff0d96385273c8c1cbfbf/devspaces-code/branding/css/codicon.css#L2) contains the corresponding style with the `devspaces` ending
- the file name (`dev-spaces.svg`) of the icon [is defined](https://github.com/redhat-developer/devspaces-images/blob/ec9c9776a0f82093616ff0d96385273c8c1cbfbf/devspaces-code/branding/css/codicon.css#L3) in the style
